### PR TITLE
Filter modified packs by marketplace version 

### PR DESCRIPTION
**Pull request by moishce**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://github.com/demisto/etc/issues/44350)

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CiscoASA/pack_metadata.json [hunk 1]
```diff
@@ -14,7 +14,6 @@
     "useCases": [],
     "keywords": [],
     "marketplaces": [
-        "xsoar",
-        "marketplacev2"
+        "xsoar"
```
Comment (ShahafBenYakir) [inline Packs/CiscoASA/pack_metadata.json:17]: Is this for testing?

File: Tests/scripts/utils/content_packs_util.py [hunk 2]
```diff
@@ -41,14 +42,32 @@ def is_pack_deprecated(pack_path: str) -> bool:
...
+    """Checks the supported marketplace versions.
+
+    Args:
+        pack_name (str): The pack name
+        id_set (dict): Structure which holds all content entities to extract pack names from.
+
+    Returns:
+        list of supported marketplace versions
+    """
+    if id_set:
```
Comment (ShahafBenYakir) [inline Tests/scripts/utils/content_packs_util.py:55]: Are we expecting not to have id set in some cases?

File: Tests/scripts/utils/content_packs_util.py [hunk 3]
```diff
@@ -41,14 +42,32 @@ def is_pack_deprecated(pack_path: str) -> bool:
...
+        return []
+
+
+def should_test_content_pack(pack_name: str, marketplace_version: str, id_set: dict) -> Tuple[bool, str]:
     """Checks if content pack should be tested in the build:
         - Content pack is not in skipped packs
         - Content pack is certified
         - Content pack is not deprecated
-
+        - Content pack is not supported the marketplace_version
```
Comment (ShahafBenYakir) [inline Tests/scripts/utils/content_packs_util.py:66]:
```suggestion
        - Content pack is supported in the marketplace_version
```

File: Tests/scripts/utils/content_packs_util.py [hunk 4]
```diff
@@ -41,14 +42,32 @@ def is_pack_deprecated(pack_path: str) -> bool:
...
     Args:
         pack_name (str): The pack name to check if it should be tested
+        marketplace_version (str):
```
Comment (ShahafBenYakir) [inline Tests/scripts/utils/content_packs_util.py:69]:
```suggestion
        marketplace_version (str): the marketplace version to collect tests for ('xsoar'/'marketplacev2')
```

File: Tests/scripts/utils/content_packs_util.py [hunk 5]
```diff
@@ -62,17 +81,21 @@ def should_test_content_pack(pack_name: str) -> Tuple[bool, str]:
...
+def should_install_content_pack(pack_name: str, marketplace_version: str, id_set: dict) -> Tuple[bool, str]:
     """Checks if content pack should be installed:
         - Content pack is not in skipped packs
         - Content pack is not deprecated
+        - Content pack is not supported the marketplace_version
 
     Args:
         pack_name (str): The pack name to check if it should be tested
-
+        marketplace_version (str):
```
Comment (ShahafBenYakir) [inline Tests/scripts/utils/content_packs_util.py:97]: See my comments from the docstring for the should_test_content_pack function

File: Packs/EmailRepIO/pack_metadata.json [hunk 6]
```diff
@@ -15,7 +15,6 @@
     "useCases": [],
     "keywords": [],
     "marketplaces": [
-        "xsoar",
```
Comment (ShahafBenYakir) [inline Packs/EmailRepIO/pack_metadata.json:18]: Also a test?

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 7]
```diff
@@ -1455,10 +1464,11 @@ def create_test_file(is_nightly, skip_save=False, path_to_pack=''):
     parser.add_argument('-s', '--skip-save', type=tools.str2bool,
                         help='Skipping saving the test filter file (good for simply doing validation)')
     parser.add_argument('-p', '--changed_pack_path', type=str, help='A string representing the changed files')
+    parser.add_argument('-t', '--marketplace_version', type=str, help='Is xsoar or marketplacev2')
```
Comment (ShahafBenYakir) [inline Tests/scripts/collect_tests_and_content_packs.py:1467]: Let's try to be consistent with the rest of the build that uses the `--marketplace` convention
```suggestion
    parser.add_argument('--marketplace', type=str, help='Is xsoar or marketplacev2')
```

**Review by ShahafBenYakir — COMMENTED** [review]
Nice

File: Packs/CiscoASA/pack_metadata.json (hunk 1 above)
Comment (moishce) [reply Packs/CiscoASA/pack_metadata.json]: yes

File: Packs/EmailRepIO/pack_metadata.json (hunk 6 above)
Comment (moishce) [reply Packs/EmailRepIO/pack_metadata.json]: yes

File: Tests/scripts/utils/content_packs_util.py (hunk 2 above)
Comment (moishce) [reply Tests/scripts/utils/content_packs_util.py:55]: no, but for an unexpected case.

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 8]
```diff
@@ -1072,9 +1083,11 @@ def remove_tests_for_non_supported_packs(tests: set, id_set: dict) -> set:
 
             # We don't want to test playbooks that are not XSOAR supported, belong to the ignored packs or from
             # deprecated packs.
-            should_test, reason = should_test_content_pack(id_set_test_playbook_pack_name)
+            should_test, reason = should_test_content_pack(id_set_test_playbook_pack_name, marketplace_version, id_set)
             if not should_test:
                 tests_that_should_not_be_tested.add(f'{test}: {reason}')
+        else:
+            tests_that_should_not_be_tested.add(f'{test}: the pack of the test was not find in the id set')
```
Comment (ShahafBenYakir) [inline Tests/scripts/collect_tests_and_content_packs.py:1090]:
```suggestion
            tests_that_should_not_be_tested.add(f'{test}: the pack of the test was not found in the id set')
```

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 9]
```diff
@@ -1380,12 +1399,17 @@ def changed_files_to_string(changed_files):
...
+def create_test_file(is_nightly, skip_save=False, path_to_pack='', marketplace_version='xsoar'):
     """Create a file containing all the tests we need to run for the CI"""
     if is_nightly:
-        packs_to_install = filter_installed_packs(set(os.listdir(constants.PACKS_DIR)))
+        packs_to_install = filter_installed_packs(set(os.listdir(constants.PACKS_DIR)), marketplace_version,
+                                                  deepcopy(ID_SET))
         tests = filter_tests(set(CONF.get_test_playbook_ids()), id_set=deepcopy(ID_SET), is_nightly=True,
-                             modified_packs=set())
+                             modified_packs=set(), marketplace_version=marketplace_version)
+        if marketplace_version == 'marketplacev2':
```
Comment (ShahafBenYakir) [inline Tests/scripts/collect_tests_and_content_packs.py:1409]: Let's add a comment explaining what we did here (maybe include the issue also)

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 10]
```diff
@@ -61,6 +61,17 @@ def get_test_playbook_ids(self):
...
         return test_ids
 
+    def get_marketplacev2_ids(self):
+        marketplacev2_tests = self._conf['test_marketplacev2']
+        test_ids = []
+        pack_ids = []
+
+        for t in marketplacev2_tests:
+            test_ids.append(t['playbookID'])
+            pack_ids.append(t['packID'])
```
Comment (ShahafBenYakir) [inline Tests/scripts/collect_tests_and_content_packs.py:71]: We usually don't mention pack names in the conf.json. Do we have to have it hardcoded? why not take it from the id set?

File: Tests/conf.json [hunk 11]
```diff
@@ -5472,5 +5472,11 @@
                 "memory_threshold": 200
             }
         }
-    }
+    },
+    "test_marketplacev2": [
+        {
+          "packID": "Phishing",
```
Comment (ShahafBenYakir) [inline Tests/conf.json:5478]: I prefer not to have the pack name hardcoded

File: Tests/scripts/infrastructure_tests/test_collect_tests_and_content_packs.py [hunk 12]
```diff
@@ -411,14 +412,40 @@ class TestChangedIntegrationAndPlaybook:
     GIT_DIFF_RET = "M Packs/PagerDuty/Integrations/PagerDuty/PagerDuty.py\n" \
                    "M Packs/CommonPlaybooks/Playbooks/playbook-Calculate_Severity_By_Highest_DBotScore.yml"
 
+    def test_changed_runnable_test__unmocked_get_modified_files_on_marketplacev2(self):
+        """
+        Given:
+            - 2 Modified in packs and only PagerDuty is marketplacev2 supported.
```
Comment (ShahafBenYakir) [inline Tests/scripts/infrastructure_tests/test_collect_tests_and_content_packs.py:418]:
```suggestion
            - 2 Modified packs where only one of them - PagerDuty is marketplacev2 supported.
```

File: Tests/scripts/infrastructure_tests/test_collect_tests_and_content_packs.py [hunk 13]
```diff
@@ -411,14 +412,40 @@ class TestChangedIntegrationAndPlaybook:
     GIT_DIFF_RET = "M Packs/PagerDuty/Integrations/PagerDuty/PagerDuty.py\n" \
                    "M Packs/CommonPlaybooks/Playbooks/playbook-Calculate_Severity_By_Highest_DBotScore.yml"
 
+    def test_changed_runnable_test__unmocked_get_modified_files_on_marketplacev2(self):
```
Comment (ShahafBenYakir) [inline Tests/scripts/infrastructure_tests/test_collect_tests_and_content_packs.py:415]: What does this name mean? Not sure I understand

File: Tests/scripts/infrastructure_tests/test_collect_tests_and_content_packs.py [hunk 14]
```diff
@@ -411,14 +412,40 @@ class TestChangedIntegrationAndPlaybook:
...
+            - Test filter should not return the CommonPlaybooks pack
+        """
+        filterd_tests, content_packs = get_mock_test_list(git_diff_ret=self.GIT_DIFF_RET, marketplace_version='marketplacev2')
+        assert filterd_tests == {'PagerDuty Test'}
+        assert content_packs == {"Base", "DeveloperTools", 'PagerDuty'}
+
+    def test_changed_runnable_test__unmocked_get_modified_files_on_xsoar(self):
+        """
+        Given:
+            - 2 Modified in packs and only CommonPlaybooks is xsoar supported.
```
Comment (ShahafBenYakir) [inline Tests/scripts/infrastructure_tests/test_collect_tests_and_content_packs.py:431]:
```suggestion
            - 2 Modified packs where only one of them - CommonPlaybooks is xsoar supported.
```

**Review by ShahafBenYakir — CHANGES_REQUESTED** [review]

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 15]
```diff
@@ -1241,31 +1285,71 @@ def get_test_list_and_content_packs_to_install(files_string,
...
+            marketplace_version (str): The marketplace version were the tests will run
+
+        Return:
+             set: set of packs
+        """
+    packs_to_install = set()
+    for artifacts in id_set.values():
+        # Ignore the Packs list in the ID set
+        if isinstance(artifacts, dict):
+            break
```
Comment (ShahafBenYakir) [inline Tests/scripts/collect_tests_and_content_packs.py:1328]: I think this should be "continue" no? we don't want to exit the loop if we get to the packs entry but to skip it.
```suggestion
            continue
```

**Review by ShahafBenYakir — COMMENTED** [review]
Great work.
Few comments in addition to the comments given in our call.

File: Tests/scripts/utils/get_modified_files_for_testing.py [hunk 16]
```diff
@@ -158,6 +153,63 @@ def create_type_to_file(files_string: str) -> Dict[FileType, Set[str]]:
...
+    file_status, file_path = line.split(maxsplit=1)
+    file_status = file_status.lower()
+    # Get to right file_path on renamed
+    if file_status.startswith("r"):
+        _, file_path = file_path.split(maxsplit=1)
+    return file_status, file_path
+
+
+def get_artifact_type(file_path: str):
+    artifact_type = file_path.split('/')[2]
```
Comment (ShahafBenYakir) [inline Tests/scripts/utils/get_modified_files_for_testing.py:207]: What if the change is for a file in level 1 or 2? Shouldn't we do something like:
```suggestion
    if length(file_path.split('/')) < 2:
         return None
    artifact_type = file_path.split('/')[2]
```

**Review by ShahafBenYakir — COMMENTED** [review]
Let me know about that one comment

**Review by ShahafBenYakir — APPROVED** [review]

**Pull request merged**